base/dict.jl

```diff
@@ -189,6 +189,16 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict, [element_type=Pair{K,V}])
```

Comment:
More precisely `similar(d::Dict, element_type=eltype(d))`. The `=` sign means that the parameter is optional and it's used all over the place.